src/Extensions/S3Storage/S3StorageEgressProviderOptions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace Microsoft.Diagnostics.Monitoring.Extension.S3Storage
+{
+    /// <summary>
+    /// Egress provider options for S3 storage.
+    /// </summary>
+    internal sealed partial class S3StorageEgressProviderOptions
+    {
+        public string Endpoint { get; set; }
+
+        [Required(AllowEmptyStrings = false)]
+        public string BucketName { get; set; }
+
+        public string RegionName { get; set; }
+
+        public string AccessKeyId { get; set; }
+
+        public string SecretAccessKey { get; set; }
+
+        public string AwsProfileName { get; set; }
+
+        public string AwsProfilePath { get; set; }
+
+        public bool GeneratePreSignedUrl { get; set; }
+
+        [Range(typeof(TimeSpan), "00:01:00", "1.00:00:00")]
```

Comment:
I don't think these attributes are checked here correct? Should we remove them?

Reply:
I'd rather preserve these for now. We need to be able to generate the schema.json file while including these options (TBD on how to do that with the extensions) and we need to validate the options within the extension (NYI).